packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2177,6 +2177,20 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             ? this.summarizer.summarizeOnDemand(...args)
             : this.summaryManager.summarizeOnDemand(...args);
     };
+
+    public readonly enqueueSummarize: ISummarizer["enqueueSummarize"] = (...args) => {
+        if (this.clientDetails.type === summarizerClientType) {
+            return this.summarizer.enqueueSummarize(...args);
+        } else if (this.summaryManager !== undefined) {
+            return this.summaryManager.enqueueSummarize(...args);
+        } else {
+            // If we're not the summarizer, and we don't have a summaryManager, we expect that
+            // generateSummaries is turned off.
+            throw new Error(
```

Comment:
Should we really throw here or return an error saying this client cannot summarize? I am thinking if I want to summarize on demand, its not straightforward to figure out if this is the summarizer client or the client that spawned the summarizer.
On the other hand, on getting fail fast, users would immediately know something went wrong and fix this.

Reply:
I think throwing is fine for now, but down the road we can add functionality here. Something like: spawn a summarizer client (maybe temporarily) if not already running.

Reply:
The problem with returning an error, is that we don't really build error results unless we are scoped to the Summarizer code. We could expand that, but I feel like fail-fast makes sense here until we add functionality. i.e. it's up to them to figure out that they are misusing the API vs. an error that might occur even if they are using the API correctly.

Reply:
Yeah that makes sense. Can you add this in a comment so that users know what to do in case they hit this failure.